Modernize Parliament

blarghmatey opened the pull request:
This udpates the Parliament repository to support Python 3.10 through 3.14, as well as moving to use uv and hatch for build/publish, and replacing use of nose with pytest. This also updates the IAM policy JSON with the latest as of 2025-10-11

blarghmatey committed:
Migrate from pkg_resources to importlib.resources for Python 3.13+ support

- Replace deprecated pkg_resources with importlib.resources.files()
- Update Python version support to 3.10-3.14 (drop 3.6-3.9)
- Update .travis.yml test matrix to Python 3.10-3.14
- Add python_requires constraint: >=3.10,<3.15
- Use proper context managers for resource file handling

This migration ensures compatibility with Python 3.13+ where pkg_resources
is no longer available. The importlib.resources API is the modern,
maintained replacement introduced in Python 3.9 and stable in 3.10+.

blarghmatey committed:
chore: Convert to UV and Hatch

blarghmatey committed:
fix: Update dependencies to compatible version

blarghmatey committed:
fix: Update tests to use pytest

blarghmatey committed:
chore: Updated IAM policy definitions and fixed tests

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/duo-labs/parliament?pullRequest=258) <br/>All committers have signed the CLA.

iainelder commented:
I'd like to be wrong, but I think this project has been abandoned. See the lack of attention in https://github.com/duo-labs/parliament/issues/247

It's great that you've put in the work to update the project for modern Python, but it might need a fork.

I haven't used Parliament for a long time but I remain subscribed to its PRs. The next time I need it, I'd like to know that it still works!

blarghmatey commented:
I agree that this project seems to have been abandoned, in which case I will continue to maintain my fork as I use this fairly heavily in https://github.com/mitodl/ol-infrastructure/. This is my good faith attempt to contribute back to the project, but in the meantime I am using my fork installed from git.

blarghmatey committed:
chore: Rebrand to ol-parliament, fix syntax warnings, refresh IAM definitions

- Rename package to 'ol-parliament' for PyPI publishing
- Update pyproject.toml: new name, authors (MIT Open Learning), correct URLs,
  sorted classifiers, explicit wheel package path for hatch
- Fix SyntaxWarning in statement.py: convert bare string regex literals with
  invalid escape sequences (\d, \*) to raw strings (r"...")
- Fix utils/update_iam_data.py: remove Chrome User-Agent header that caused
  405 responses from AWS docs; replace deprecated findAll() with find_all()
- Regenerate parliament/iam_definition.json from current AWS docs (469 services)
- Modernize .github/workflows/publish.yml: use uv build, OIDC trusted
  publishing via pypa/gh-action-pypi-publish, actions pinned to v4/v5
- Add .pi-lens/ to .gitignore

blarghmatey commented:
Since this PR hasn't gotten any feedback I have published my fork as `ol-parliament` on PyPI and can be tracked at https://github.com/mitodl/ol-parliament. I will likely sever the fork relationship at some point but haven't done that yet so that it's still discoverable in this project's fork network.
